In [34]:
import os
import numpy as np
import pandas as pd
import sklearn as sk

# Set the working directory
root_dir = "Datasets"

# Read dataset and save it in a dataframe
df = pd.read_csv(os.path.join(root_dir, "dermatology.data"), header=None, delimiter=",")
print(df.shape)


(366, 35)


### Dataset review

- **Dataset presentation:** The dataset consists of clinical cases of patients that suffer from erythemato-squamous diseases.

- **File conversion:** We did not need to convert our plain text, we used the read_csv function from pandas.

- **Entry and attribute count:** The dataset consists of 366 entries, each of which has 34 attributes. All of them are ordinal, except the family history which is binary and age which is numeric.

- **Indexes and headers:** There are no row indexes or column headers.

- **Class labels:** Class labels (found in column 35) are integers from 1 to 6, that represent psoriasis, seboreic dermatitis, lichen planus,pityriasis rosea, cronic dermatitis, and pityriasis rubra pilaris respectively.

- **Missing values:** There are 8 missing values in the age column and they represent 8/366= 0.022% of the entries.

- **Class count:** As we can see from the table below, the dataset is not balanced.

    Class      | Number of instances | Percentages |
    ----------- | ----------- | ----------- |
    psoriasis      | 112       | 30.6% |
    seboreic dermatitis   | 61        | 16.6% |
    lichen planus         | 72       | 19.6% |
    pityriasis rosea   | 49        | 13.4% |
    cronic dermatitis         | 52       | 14.3% |
    pityriasis rubra pilaris    | 20        | 5.5% |


      


